compiler/fir/analysis-tests/testData/resolve/cfg/flowFromInplaceLambda2.kt

```diff
@@ -0,0 +1,66 @@
+// !DUMP_CFG
+
+fun <T> foo(x: T?, i: Int, y: T) {}
+
+fun <T> id(x: T): T = x
+
+fun <T> n(): T? = null
+
+fun someCompletedCall(arg: Int) = arg
+
+fun test1(x: String?) {
+    foo(
+        id(run { x as String; n() }),
+        1,
+        run { x<!UNSAFE_CALL!>.<!>length; 123 } // Bad (resolution order undefined)
+    )
+    x.length // OK (x as String unconditional)
+}
+
+fun test2(x: String?) {
+    foo(
+        id(run { x as String; n() }),
+        someCompletedCall(1),
+        run { x<!UNSAFE_CALL!>.<!>length; 123 } // Bad (resolution order undefined)
+    )
+    x.length // OK (x as String unconditional)
+}
+
+fun test3(x: String?) {
+    foo(
+        id(run { x as String; n() }),
+        if (true) 1 else 2,
+        run { x<!UNSAFE_CALL!>.<!>length; 123 } // Bad (resolution order undefined)
+    )
+    x.length // OK (x as String unconditional)
+}
+
+fun test4(x: String?) {
+    var p = x
+    if (p != null) {
+        foo(
+            id(if (true) run { p = null; n() } else { n() }),
+            1,
+            run { p<!UNSAFE_CALL!>.<!>length; 123 } // Bad (p = null possible)
+        )
+        p<!UNSAFE_CALL!>.<!>length // Bad (p = null possible)
+    }
+}
+
+fun test5(x: String?, y: String?) {
+    foo(
+        y?.let { x as String; n() },
+        1,
+        run { "" }
+    )
+    x<!UNSAFE_CALL!>.<!>length // Bad (x as String conditional)
+}
+
+fun test6(x: String?) {
+    foo(
+        id(if (true) run { x as String; n() } else run { x as String; n() }),
+        1,
+        run { x<!UNSAFE_CALL!>.<!>length; 123 } // Bad (resolution order undefined)
+    )
+    x<!UNSAFE_CALL!>.<!>length // OK (x as String in both branches)
```

Comment:
This doesn't work because the DFG for ifs is not entirely correct: it joins flow from both lambdas and the if itself, but the if does not have type information from the lambdas, so here everything done by lambdas is lost. Not any worse than the old behavior, but ideally flow should be joined from lambdas and all branches that don't end in lambdas.